lib/filestorage.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const common = require('metarhia-common');
+const utils = require('./utils.js');
+
+class FileStorage {
+
+  // Create new FileStorage
+  //   options - <object>
+  //     dir - <string>, data storage directory
+  //     minCompressSize - <number>, minimal file size
+  //                       to be compressed, default = 1024
+  constructor(options) {
+    this.dir = path.resolve(options.dir);
+    this.minCompressSize = options.minCompressSize || 1024;
+  }
+
+  // Write file to storage
+  //   id - <Object>, common.Uint64, id of file
+  //   data - <string> | <Buffer> | <Uint8Array>, data to be written
+  //   opts - <Object>, optional
+  //     checksum - <string>, checksum type
+  //     dedupHash - <string>, second checksum type
+  //   cb - <Function>, callback
+  //     err - <Error>
+  //     checksum - <string>, data checksum
+  //     dedupHash - <string>, second data checksum
+  //     originalSize - <number>, data size
+  // Throws: <TypeError>, if `opts.checksum` or `opts.dedupHash` is incorrect
+  write(id, data, opts = {}, cb) {
+    if (typeof opts === 'function') {
+      cb = opts;
+      opts = {};
+    }
+    const checksum = utils.computeHash(data, opts.checksum || 'CRC32');
```

Comment:
I'd prefer to move the default checksum strategy into a package constant, same for `dedupHash`.